### ISSUE TITLE
The CLI silently fails when attempting to use an invalid command

### ISSUE THREAD

**Issue by nihalgonsalves** [pinterest/ktlint#629]
There is no response for invalid commands, nor any indication via the exit code:

```
$ ktlint -V
0.35.0

$ ktlint foo

$ echo $? 
0  
```

**Expected Behaviour:**

A typo or invalid command should generate an error message and return with a non-zero exit code.

**Comment by Tapchicoma** [comment]
In your example `foo` is file pattern ktlint uses to find `*.kt` files. If you will run with `./ktlint --foo` then it will fail execution with `Error: Unmatched argument [--foo]` and general help.

Imho exit code `0` is valid in `./ktlint foo` case as it means no Kotlin files found and everything is ok.

**Comment by nihalgonsalves** [comment]
@Tapchicoma Thanks for the explanation. However, I still believe that the exit code isn't valid as top level commands (`installGitPreCommitHook`, `installGitPrePushHook`, `printAST`, `applyToIDEA`, `applyToIDEAProject`) exist. Misspelling one of these should not result in a successful exit.

I see why it isn't an easy fix. One option to inform users about a misspelling or invalid command could be to print a warning such as "no files matched" to stderr.
